### ISSUE TITLE
"Hook can only be invoked from render methods" on Next.js + Preact

### ISSUE THREAD

**Issue by aaronlin0122** [nanostores/preact#4]
Using next-plugin-preact :

![image](https://user-images.githubusercontent.com/47508893/190864872-14480099-44a1-48fd-a115-1b516e9ba94d.png)

**Comment by ai** [comment]
Can you show how do you use `Counter` and the full code of `Counter` file?

**Comment by aaronlin0122** [comment]
> Can you show how do you use `Counter` and the full code of `Counter` file?

`components/Counter.tsx`
```tsx
import React from 'react'
import { MdAdd, MdRemove } from 'react-icons/md'
import { useStore } from '@nanostores/preact'
import { _counter, increment, decrement } from '@/store/counter'

function Counter() {
  const counter = useStore(_counter)

  return (
    <div className='mt-4 flex items-center'>
      <button
        type='button'
        title='Decrement'
        className='rounded-full bg-white p-1 text-2xl text-gray-800 opacity-70'
        onClick={() => decrement()}
      >
        <MdRemove className='block' />
      </button>
      <span className='mx-4 text-4xl'>{counter}</span>
      <button
        type='button'
        title='Increment'
        className='rounded-full bg-white p-1 text-2xl text-gray-800 opacity-70'
        onClick={() => increment()}
      >
        <MdAdd className='block' />
      </button>
    </div>
  )
}

export default Counter
```

`store/counter.ts`
```ts
import { atom } from 'nanostores'

export const _counter = atom(0)

export function increment() {
  _counter.set(_counter.get() + 1)
}

export function decrement() {
  _counter.set(_counter.get() - 1)
}
```

**Comment by aaronlin0122** [comment]
You can check the template in here https://github.com/NekoChanTaiwan/nextjs-lite-template

**Comment by ai** [comment]
Sorry, I used Preact only a few times and have no enough experience with this bug.

**Comment by aaronlin0122** [comment]
It's ok, same as me, I just try to play around Preact and make a lightweight template for Next.js.

And I found a solution for this problem, I copied the `useStore.js` from this repo,I just replaced
`import { useState, useEffect } from 'preact/hooks'` to `import { useState, useEffect } from 'react'` ,
and it is working, I think the bug is because I'm using `next-plugin-preact` plugin for Next.js.

here is my modified version:
https://github.com/NekoChanTaiwan/nextjs-lite-template/blob/main/hooks/useStore.ts#L4

I also test the `@nanostores/preact` in Vite + Preact + TS, and It is work fine.
repo: https://github.com/NekoChanTaiwan/preact-nanostores
preview: https://preact-nanostores.vercel.app/

( sorry for my poor English 🥲
